### DISCUSSION TITLE
managing completely different dotfiles from multiple machines

### DISCUSSION THREAD

**Discussion by ayushnix** [twpayne/chezmoi#1508] [Q&A] [answered]
What would be the ideal method to manage different completely dotfiles across different machines?

I have a router with OpenWRT on it and it has its dotfiles in `/etc/config` and they are completely different from anything I have on my Arch Linux laptop. Using templates here wouldn't help because some of these files have nothing in common. Unlike what most `sshd_config` files look like, here's how the SSH config looks like on OpenWRT -

```
root@router:~# cat /etc/config/dropbear

config dropbear
        option Port '22'
        option Interface 'lan'
        option PasswordAuth 'off'
        option RootPasswordAuth 'off'
```

Here some things I'm thinking of
- Should I create an orphan branch and manage my router dotfiles on my [primary chezmoi dotfiles repo](https://github.com/ayushnix/dotfiles) itself?
- Should I create entirely different dotfile repos, one for my laptop and another for OpenWRT? 
- Should I just dump the router dotfiles inside a folder on my laptop periodically and then have chezmoi manage them as part of my primary dotfiles repo?

I'm basically looking for recommendations on which method would make the most sense to accomplish this.

**Comment by twpayne** [comment, marked as answer]
If the dotfiles really are completely different with no data shared in common, then I'd recommend either the orphan branch approach if you want to be fancy (note the [`--branch` option to `chezmoi init`](https://github.com/twpayne/chezmoi/blob/master/docs/REFERENCE.md#init-repo)) or a separate dotfiles repo if you want to be simple.

If there are some data in common between the machines, then consider either storing that shared data in your password manager or  using an [external](https://github.com/twpayne/chezmoi/blob/master/docs/REFERENCE.md#chezmoiexternalformat).

#1410 might also be of interest, comments welcome.

**Comment by ayushnix** [comment]
Hmm, I'll go ahead with orphan branch approach. Thanks!
